src/WebJobs.Script.WebHost/App_Start/AutofacBootstrap.cs

```diff
@@ -17,7 +17,7 @@ internal static void Initialize(ContainerBuilder builder, WebHostSettings settin
 
             // these services are externally owned by the WebHostResolver, and will be disposed
             // when the resolver is disposed
-            builder.Register<SecretManager>(ct => webHostResolver.GetSecretManager(settings)).ExternallyOwned();
+            builder.Register<ISecretManager>(ct => webHostResolver.GetSecretManager(settings)).ExternallyOwned();
```

Comment:
Primarily for testability, introduced the ISecretManager and updated the references throughout